src/dbnode/client/session.go

```diff
@@ -2319,11 +2319,28 @@ func (s *session) streamBlocksMetadataFromPeer(
 
 	fetchFn := func() error {
 		borrowErr := peer.BorrowConnection(checkedAttemptFn)
+		// Don't retry if the host is not available to prevent exponential
+		// backoff from causing this to take a long time when subsequent
+		// requests will fail.
+		if isHostNotAvailableError(borrowErr) {
+			borrowErr = xerrors.NewNonRetryableError(borrowErr)
+		}
 		return xerrors.FirstError(borrowErr, attemptErr)
 	}
 
 	for moreResults {
 		if err := s.streamBlocksRetrier.Attempt(fetchFn); err != nil {
+			// Check if the error was a hostNotAvailableError wrapped in a
+			// nonRetryableError. If so, return the hostNotAvailableError
+			// instead of the non-retryable error since the error is technically
+			// retryable, it was only wrapped to prevent the exponential backoff
+			// in the actual retrier.
```

Comment:
Maybe just make the `newHostNotAvailableError(...)` return a nonRetryableError?  

i.e. 
```go
func newHostNotAvailableError(err error) error {
	return xerrors.newNonRetryableError(hostNotAvailableError{err: err})
}

func isHostNotAvailableError(err error) bool {
  inner := xerrors.GetInnerNonRetryableError(err)
  if inner == nil {
    return false
  }
  _, ok := inner.(hostNotAvailableError)
  return ok
}
```